### PR TITLE
Fix crates_crawler

### PR DESCRIPTION
**Pull request by timgluz**
This PR:

* fixes failing tests in the `crates_crawler_spec`
* fixes URL of crates page;
* replaces `Project::A_TYPE_CRATES` with `Project::A_TYPE_CARGO` to remove confision between website and project format.

### CODE REVIEW

File: lib/versioneye/crawlers/crates_crawler.rb [hunk 1]
```diff
@@ -194,9 +197,11 @@ def self.upsert_product_owner(product_db, owner_doc, version_num)
       language: product_db[:language],
       prod_key: product_db[:prod_key],
       version: version_num,
-      name: owner_id
+      developer: owner_id
```
Comment (reiz) [inline lib/versioneye/crawlers/crates_crawler.rb]: @timgluz Can you revert this change please? `developer` is kind of legacy. The real attribute which counts is the `name`. I documented that with a comment in the source code.

File: lib/versioneye/crawlers/crates_crawler.rb [hunk 2]
```diff
@@ -284,10 +289,10 @@ def self.upsert_version_link(product_db, version_id, name, url)
       language: product_db[:language],
       prod_key: product_db[:prod_key],
       version_id: version_id,
-      link: url
+      name: name
```
Comment (reiz) [inline lib/versioneye/crawlers/crates_crawler.rb]: @timgluz Please revert that to `link`. The `name` of a link can be empty. The `link` itself is mandatory.

File: lib/versioneye/crawlers/crates_crawler.rb [hunk 3]
```diff
@@ -284,10 +289,10 @@ def self.upsert_version_link(product_db, version_id, name, url)
...
     ).first_or_create
 
-    url_db.update(name: name)
+    url_db.update(link: url)
```
Comment (reiz) [inline lib/versioneye/crawlers/crates_crawler.rb]: @timgluz Revert this change. See comments above.

File: lib/versioneye/crawlers/crates_crawler.rb (hunk 1 above)
Comment (timgluz) [inline lib/versioneye/crawlers/crates_crawler.rb]: Hmm, should we keep author's full name somewhere? or it comes from somewhere else?

File: lib/versioneye/crawlers/crates_crawler.rb (hunk 2 above)
Comment (timgluz) [inline lib/versioneye/crawlers/crates_crawler.rb]: Ok, i see that we wouldnt have duplicated URLs.

File: lib/versioneye/crawlers/crates_crawler.rb (hunk 1 above)
Comment (reiz) [inline lib/versioneye/crawlers/crates_crawler.rb]: @timgluz Good questions. I would say we have to add another field to the developers collection. A field for `fullname`.
